client/src/modules/list/components/ItemsContainer.jsx

```diff
@@ -1,94 +1,46 @@
-import { SortOrderType } from 'common/constants/enums';
-import FilterBox from 'modules/list/components/FilterBox';
 import { getCurrentUser } from 'modules/user/model/selectors';
-import { IntlPropType, UserPropType } from 'common/constants/propTypes';
+import { IntlPropType } from 'common/constants/propTypes';
 import './ItemsContainer.scss';
+import Filter from 'common/components/Filter';
+import { CloseIcon } from 'assets/images/icons';
 
-const SortOptionType = Object.freeze({
-  NAME: 'name',
-  DATE: 'createdAt',
-  AUTHOR: 'author',
-  VOTES: 'votes'
-});
-
-export const FilterOptionType = Object.freeze({
-  MY_ITEMS: 'my_items',
-  ALL_ITEMS: 'all_items'
-});
+class ItemsContainer extends Component {
+  constructor(props) {
+    super(props);
 
-const sortOptions = [
-  { id: SortOptionType.AUTHOR, label: 'list.sort-box.author' },
-  { id: SortOptionType.DATE, label: 'list.sort-box.createdAt' },
-  { id: SortOptionType.NAME, label: 'list.sort-box.name' },
-  { id: SortOptionType.VOTES, label: 'list.sort-box.votes' }
-];
+    const { items } = this.props;
 
-const filterOptions = [
-  { id: FilterOptionType.ALL_ITEMS },
-  { id: FilterOptionType.MY_ITEMS }
-];
+    this.state = {
+      items
+    };
+  }
 
-class ItemsContainer extends Component {
-  state = {
-    sortBy: SortOptionType.DATE,
-    sortOrder: SortOrderType.DESCENDING,
-    filterBy: FilterOptionType.ALL_ITEMS
-  };
+  componentDidUpdate(previousProps) {
+    const { items } = this.props;
+    const { items: previousItems } = previousProps;
 
-  onSortChange = (sortBy, sortOrder) => this.setState({ sortBy, sortOrder });
-
-  sortItems = (items, sortBy, sortOrder) => {
-    let result = [...items];
-
-    switch (sortBy) {
-      case SortOptionType.NAME:
-        result = _sortBy(result, item => item.name.toLowerCase());
-        break;
-      case SortOptionType.AUTHOR:
-        result = _sortBy(result, [
-          item => item.authorName.toLowerCase(),
-          item => item.name.toLowerCase()
-        ]);
-        break;
-      case SortOptionType.DATE:
-        result.sort((a, b) => {
-          const dateA = new Date(a.createdAt);
-          const dateB = new Date(b.createdAt);
-
-          return dateA - dateB;
-        });
-        break;
-      case SortOptionType.VOTES:
-        result.sort((a, b) => a.votesCount - b.votesCount);
-        break;
-      default:
-        break;
+    if (!_isEqual(items, previousItems)) {
+      this.updateItems();
     }
+  }
 
-    return sortOrder === SortOrderType.ASCENDING ? result : result.reverse();
-  };
-
-  onFilterChange = filterBy => this.setState({ filterBy });
-
-  filterItems = (items, filterBy) => {
-    const {
-      currentUser: { id }
-    } = this.props;
+  updateItems = () => {
+    const { items } = this.props;
 
-    return filterBy === FilterOptionType.MY_ITEMS
-      ? items.filter(item => item.authorId === id)
-      : items;
+    this.setState({ items });
```

Comment:
Why not pure render filtering? Are there any performance issues present?

Reply:
It was a customer's wish to replace the filter and sort components with search input

Reply:
This is rather a technical question as this should not affect the UI. I think you should strive to minimize state and if possible, use pure filtering instead.

Reply:
Sorry, but I don't understand what you mean by pure filtering?